File: 백준/Silver/15650. N과 M （2）/N과 M （2）.cc

#include <iostream>
using namespace std;

int arr[9];
int n, m;

void dfs(int cnt, int start) {
    if (cnt == m) { // m개를 뽑았으면 출력
        for (int i = 0; i < m; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
        return;
    }
    for (int i = start; i <= n; i++) {
        arr[cnt] = i;         // 현재 수 선택
        dfs(cnt + 1, i + 1);  // 다음 수는 i보다 큰 수만 선택
    }
}

int main() {
    cin >> n >> m;
    dfs(0, 1); // 1부터 시작
}

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py

def dfs(start,depth):

    #개수에 맞게 꽉 찼다면 결과 출력
    if depth==m:
        print(' '.join(map(str, selected)))
        return

    #Start부터 n까지 반복
    for i in range(start,n+1):
        selected[depth]=i
        dfs(i+1,depth+1)




n,m=map(int,input().split())

selected=[0]*m

dfs(1,0)
